includes/plugins/woocommerce/my-account/templates/v1/account-settings.php

```diff
@@ -0,0 +1,276 @@
+			<input type="email" class="woocommerce-Input woocommerce-Input--email input-text" name="account_email_display" id="account_email_display" autocomplete="email" disabled value="<?php echo \esc_attr( $user->user_email ); ?>" />
+			<span class="legend">
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// Translators: %s is the reply-to email address for the site.
+						__( 'To update your email address, please <a href="mailto:%s">contact us</a>.', 'newspack-plugin' ),
+						Emails::get_reply_to_email()
+					)
+				);
+				?>
+			</span>
+			<input type="hidden" class="woocommerce-Input woocommerce-Input--email input-text" name="account_email" id="account_email" autocomplete="email" value="<?php echo \esc_attr( $user->user_email ); ?>" />
+			<?php endif; ?>
+		</p>
+
+		<?php
+			/**
+			 * Hook where additional fields should be rendered.
+			 *
+			 * Newspack equivalent of do_action( 'woocommerce_edit_account_form_fields' );
+			 */
+			do_action( 'newspack_woocommerce_edit_account_form_fields' );
+		?>
+
+		<?php
+			/**
+			 * My Account edit account form.
+			 *
+			 * Newspack equivalent of do_action( 'woocommerce_edit_account_form' );
+			 */
+			\do_action( 'newspack_woocommerce_edit_account_form' );
+		?>
+
+		<p class="woocommerce-buttons-card">
+			<?php \wp_nonce_field( 'save_account_details', 'save-account-details-nonce' ); ?>
+			<?php if ( $is_email_change_enabled && $is_pending_email_change ) : ?>
+				<a href="<?php echo esc_url( WooCommerce_My_Account::get_email_change_url( WooCommerce_My_Account::CANCEL_EMAIL_CHANGE_PARAM, $user->user_email ) ); ?>" class="woocommerce-Button button ma0"><?php \esc_html_e( 'Cancel email change', 'newspack-plugin' ); ?></a>
+			<?php endif; ?>
+			<input type="hidden" name="action" value="save_account_details" />
+			<button type="submit" class="woocommerce-Button button primary newspack-ui__button--wide-on-mobile" name="save_account_details" value="<?php \esc_attr_e( 'Save changes', 'newspack-plugin' ); ?>"><?php \esc_html_e( 'Update profile', 'newspack-plugin' ); ?></button>
+		</p>
+
+		<?php \do_action( 'newspack_woocommerce_edit_account_form_end' ); ?>
+	</form>
+</section>
+
+<section id="account-password">
+	<h1 class="newspack-ui__font--m"><?php \esc_html_e( 'Password', 'newspack-plugin' ); ?></h1>
+	<form method="post" class="woocommerce-ResetPassword lost_reset_password">
+		<?php if ( ! $without_password ) : ?>
+		<p class="woocommerce-form-row woocommerce-form-row--wide form-row form-row-wide">
+			<label for="current_password"><?php esc_html_e( 'Current password', 'newspack-plugin' ); ?>&nbsp;<span class="required" aria-hidden="true">*</span><span class="screen-reader-text"><?php esc_html_e( 'Required', 'newspack-plugin' ); ?></span></label>
+			<input
+				type="password"
+				class="woocommerce-Input woocommerce-Input--text input-text"
+				name="current_password"
+				id="current_password"
+				required
+				aria-required="true"
+			/>
+		</p>
+		<?php endif; ?>
```

Comment:
I'm still working on my review, but I'd like to point this out before going through everything.

For security reasons, we shouldn't allow password changes without re-authentication. If the password has not yet been set, I suggest we make the reader go through the regular password reset flow via email (what we have now)

Reply:
Good point, and I also realize I missed a custom modal to actually reset the password in the email-based flow, too. Will work on a fix for this flow

Reply:
98bab4d8 forces the user to go through the email-based flow if they've never set a password before:

<img width="394" alt="Screenshot 2025-05-21 at 4 07 47 PM" src="https://github.com/user-attachments/assets/84ad998c-1849-4034-8a9f-beb7b15a29a1" />

Digging in a little, it's more complicated to bypass Woo's default "reset password" form since it's handled via redirects. I might work on that in a separate PR

Reply:
46ce0a4b should do it, actually. 31cf3df and 663684f also ensure that the password reset key check from core Woo still works (so that you can't use the same "Reset password" link more than once). Updated the testing instructions.